examples/keras_io/tensorflow/generative/cyclegan.py

```diff
@@ -0,0 +1,667 @@
+# Gamma initializer for instance normalization.
+gamma_init = keras.initializers.RandomNormal(mean=0.0, stddev=0.02)
+
+buffer_size = 256
+batch_size = 1
+
+
+def normalize_img(img):
+    img = ops.cast(img, dtype="float32")
+    # Map values in the range [-1, 1]
+    return (img / 127.5) - 1.0
+
+
+def preprocess_train_image(img, label):
+    # Random flip
+    img = tf_image.random_flip_left_right(img)
+    # Resize to the original size first
+    img = tf_image.resize(img, [*orig_img_size])
+    # Random crop to 256X256
+    img = tf_image.random_crop(img, size=[*input_img_size])
+    # Normalize the pixel values in the range [-1, 1]
+    img = normalize_img(img)
+    return img
+
+
+def preprocess_test_image(img, label):
+    # Only resizing and normalization for the test images.
+    img = tf_image.resize(img, [input_img_size[0], input_img_size[1]])
+    img = normalize_img(img)
+    return img
+
+
+"""
+## Create `Dataset` objects
+"""
+
+
+# Apply the preprocessing operations to the training data
+train_horses = (
+    train_horses.map(preprocess_train_image, num_parallel_calls=autotune)
+    .cache()
+    .shuffle(buffer_size)
+    .batch(batch_size)
+)
+train_zebras = (
+    train_zebras.map(preprocess_train_image, num_parallel_calls=autotune)
+    .cache()
+    .shuffle(buffer_size)
+    .batch(batch_size)
+)
+
+# Apply the preprocessing operations to the test data
+test_horses = (
+    test_horses.map(preprocess_test_image, num_parallel_calls=autotune)
+    .cache()
+    .shuffle(buffer_size)
+    .batch(batch_size)
+)
+test_zebras = (
+    test_zebras.map(preprocess_test_image, num_parallel_calls=autotune)
+    .cache()
+    .shuffle(buffer_size)
+    .batch(batch_size)
+)
+
+
+"""
+## Visualize some samples
+"""
+
+
+_, ax = plt.subplots(4, 2, figsize=(10, 15))
+for i, samples in enumerate(zip(train_horses.take(4), train_zebras.take(4))):
+    horse = (((samples[0][0] * 127.5) + 127.5).numpy()).astype(np.uint8)
+    zebra = (((samples[1][0] * 127.5) + 127.5).numpy()).astype(np.uint8)
+    ax[i, 0].imshow(horse)
+    ax[i, 1].imshow(zebra)
+plt.show()
+
+
+"""
+## Building blocks used in the CycleGAN generators and discriminators
+"""
+
+
+class ReflectionPadding2D(layers.Layer):
+    """Implements Reflection Padding as a layer.
+
+    Args:
+        padding(tuple): Amount of padding for the
+        spatial dimensions.
+
+    Returns:
+        A padded tensor with the same type as the input tensor.
+    """
+
+    def __init__(self, padding=(1, 1), **kwargs):
+        self.padding = tuple(padding)
+        super().__init__(**kwargs)
+
+    def call(self, input_tensor, mask=None):
+        padding_width, padding_height = self.padding
+        padding_tensor = ([
+            [0, 0],
+            [padding_height, padding_height],
+            [padding_width, padding_width],
+            [0, 0],
+        ])
+        return ops.pad(input_tensor, ops.convert_to_tensor(padding_tensor), mode='reflect')
```

Comment:
Make sure to run `black` on the file to format the code

Reply:
Done. Thanks.